src/test/java/com/google/songgame/YoutubeParserTest.java

```diff
@@ -0,0 +1,102 @@
+package com.google.songgame.data;
+
+import org.junit.Assert;
+import org.junit.rules.ExpectedException;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.Rule;
+import org.junit.Ignore;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class YoutubeParserTest {
+
+  private YoutubeParser youtubeParser;
+
+  @Rule public ExpectedException thrown = ExpectedException.none();
+
+  @Before
+  public void init() {
+    youtubeParser = new YoutubeParser();
+  }
+
+  // Test getPlaylistIdFromUrl
+  @Test
+  public void parsePlaylistUrlWithParameters() {
+    String testPlaylistUrl =
```

Comment:
Is there any differenced between the first and second thing you're testing in this method, aside from the playlist ID? If not, I would delete the second part.

When testing, you usually want to handle all of the different cases/code paths, and I think you've done this well here with your different methods, e.g. handling a standard link, a share link, and non-playlist link, etc. However, there isn't a need to test that multiple different playlist IDs work within the same test case. If the code can extract a playlist ID from a standard link, it can also extract a slightly different playlist ID from the same link structure.

This may apply to some of your other tests as well, so please clean those up too.